### ISSUE TITLE
override regex parsing failure on Windows

### ISSUE THREAD

**Issue by pballok** [vladopajic/go-test-coverage#115]
Hi there,
Thanks for this repo, it is just what I needed :)

I think I found a small issue with how regex is parsed for the `override` (I am running this on Windows)
This is the config I am running:
```
override:
  - threshold: 50
    path: ^internal/graph
```
When I run your tool, I get the following error:
```
panic: regexp: Compile(`^internal\graph`): error parsing regexp: invalid escape sequence: `\g`
```
It looks like the path delimiter got switched from '/' to '\\' (which makes sense on Windows) but then the '\\' got interpreted as a special character of the regex.
If this is indeed the case, the '/' should be replaced with '\\\\' on Windows.

**Comment by pballok** [comment]
I just checked deeper in your code, and I see you're replacing the `/` character with `filepath.Separator`, and this should be ok, at least according to [this](https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/os/path_windows.go;l=14)  in go 1.23 the Separator on Windows should already be `\\\\`. So maybe I have a different problem.

**Comment by vladopajic** [comment]
@pballok hey, thanks for reporting issue.

yes it appears that there is issue with this logic for windows. `\` will escape special character, but since `g` is not special character it panics (it actually is, but not supported by go). 

as dirty workaround, i think `^internal//graph`  this should work for you. could you check ?

**Comment by pballok** [comment]
I tried that already, but unfortunately it doesn't work. There is no panic any more, but the pattern doesn't match the files and packages it should. For reference, here is the entire config:
```
# (mandatory)
# Path to coverprofile file (output of `go test -coverprofile` command).
#
# For cases where there are many coverage profiles, such as when running
# unit tests and integration tests separately, you can combine all those
# profiles into one. In this case, the profile should have a comma-separated list
# of profile files, e.g., 'cover_unit.out,cover_integration.out'.
profile: cover.out

# (optional; but recommended to set)
# When specified reported file paths will not contain local prefix in the output
local-prefix: "github.com/pballok/gurps-bchest-be"

# Holds coverage thresholds percentages, values should be in range [0-100]
threshold:
  file: 70
  package: 80
  total: 90

# Holds regexp rules which will override thresholds for matched files or packages
# using their paths.
#
# First rule from this list that matches file or package is going to apply
# new threshold to it. If project has multiple rules that match same path,
# override rules should be listed in order from specific to more general rules.
override:
  - threshold: 50
    path: ^internal//graph

# Holds regexp rules which will exclude matched files or packages
# from coverage statistics
exclude:
  paths:
    - generated\.go$
    - _gen\.go$
    - mock_.*\.go$
    - filesystem\.go$
```
And here is the output of the script. Note the threshold of both the `internal/graph` package and the `internal/graph/schema.resolvers.go` still have the default threshold, the override wasn't applied:
```
go test ./internal/... -coverprofile=./cover.out -covermode=count
        github.com/pballok/gurps-bchest-be/internal/graph/model         coverage: 0.0% of statements
        github.com/pballok/gurps-bchest-be/internal/graph               coverage: 0.0% of statements
ok      github.com/pballok/gurps-bchest-be/internal/attribute   0.228s  coverage: 100.0% of statements
ok      github.com/pballok/gurps-bchest-be/internal/character   0.235s  coverage: 100.0% of statements
ok      github.com/pballok/gurps-bchest-be/internal/memstorage  0.226s  coverage: 100.0% of statements
ok      github.com/pballok/gurps-bchest-be/internal/server      0.121s  coverage: 50.0% of statements
ok      github.com/pballok/gurps-bchest-be/internal/storage     0.258s  coverage: 35.2% of statements
go run github.com/vladopajic/go-test-coverage/v2 --config=./.testcoverage.yml
File coverage threshold (70%) satisfied:        FAIL
  below threshold:                              coverage:       threshold:
  internal/graph/schema.resolvers.go            0%              70%
  internal/server/server.go                     50%             70%

Package coverage threshold (80%) satisfied:     FAIL
  below threshold:                              coverage:       threshold:
  internal/graph                                0%              80%
  internal/server                               50%             80%

Total coverage threshold (90%) satisfied:       FAIL
Total test coverage: 81%
```

**Comment by pballok** [comment]
I cloned your repo and did a bit of investigating, here is what I found:

When the file names and package names are matched against the override path regexp, the regexp has the os-specific path separator, but the file names and package names don't. This is why even after getting around the `\` special character problem, the override path regex never matches any of the file name.

In fact, when I removed the call to the `path.NormalizePathInRegex` from the override paths, everything started to work fine, neither the regexp, nor the file names had the windows-stly separators, so the regexp was successfully matching.

If you prefer, I can try to come up with a PR, if that helps.
